flink-core/src/main/java/org/apache/flink/configuration/NetworkEnvironmentOptions.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.configuration;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.annotation.docs.ConfigGroup;
+import org.apache.flink.annotation.docs.ConfigGroups;
+
+import static org.apache.flink.configuration.ConfigOptions.key;
+
+/**
+ * The set of configuration options relating to network stack.
+ */
+@PublicEvolving
+@ConfigGroups(groups = @ConfigGroup(name = "NetworkEnvironment", keyPrefix = "network.environment"))
+public class NetworkEnvironmentOptions {
```

Comment:
As I understood, we could move this class also to `org.apache.flink.runtime.io.network` if `org.apache.flink.docs.configuration.ConfigOptionsDocGenerator#LOCATIONS` is configured properly like it was for `NettyConfig` by including `org.apache.flink.runtime.io.network.netty` into `LOCATIONS` (which should be removed now).

Reply:
If we regard `NetworkEnvironmentOptions` as general one which could be used for other shuffle services future, then it makes sense to put it under configuration package like other options. And the `org.apache.flink.runtime.io.network.netty` could be removed. 

If we think of it as `NetworkEnvironment` private, it could be put under `network` package. And `org.apache.flink.runtime.io.network.netty` could be changed to `org.apache.flink.runtime.io.network`.

What do you think?

Reply:
ok, looking more how other options are added, if we say that the network environment shuffle is the default one, it can also stay in the core package, agreed :)